tests/publiccloud/migration.pm

```diff
@@ -21,42 +21,117 @@ use utils;
+    my $act_out = `curl -s $act_url`;
+    $act_rpm = $1 if $act_out =~ /($act_pkg.*?rpm)/g;
+    my $pc_out = `curl -s $pc_url`;
+    $pc_rpm = $1 if $pc_out =~ /($pc_pkg.*?rpm)/g;
+    my $dms_act_rpm = "$act_url" . "$act_rpm";
+    my $dms_pc_rpm  = "$pc_url" . "$pc_rpm";
+    my $tmp_repo = "/tmp/sles15-mig-repo";
+    my $sle15_mig_repo = "/tmp/SLE15_Migration";
+
+    record_info( "dms package name", $dms_rpm_name );
+    record_info( "act package name", $act_rpm );
+    record_info( "pc package name",  $pc_rpm );
+
+    record_info('wget repokey and rpm');
+    assert_script_run( "wget $dms_repo_key_url -O /tmp/$dms_repo_key", 180 );
+    assert_script_run( "wget $dms_rpm -O /tmp/$dms_rpm_name",          180 );
+    assert_script_run( "wget $dms_act_rpm -O /tmp/$act_rpm",           180 );
+    assert_script_run( "wget $dms_pc_rpm -O /tmp/$pc_rpm",             180 );
+
+    record_info('SCP repokey and rpm');
+
+    #Download scp the rpm packages and repokey to public cloud
+    my $repo_key =
+      $instance->scp( "/tmp/$dms_repo_key", 'remote:' . "/tmp/$dms_repo_key", 200 );
+    my $dms_rpm =
+      $instance->scp( "/tmp/$dms_rpm_name", 'remote:' . "/tmp/$dms_rpm_name", 1000 );
+    my $dms_act_rpm =
+      $instance->scp( "/tmp/$act_rpm", 'remote:' . "/tmp/$act_rpm", 1000 );
+    my $dms_pc_rpm =
+      $instance->scp( "/tmp/$pc_rpm", 'remote:' . "/tmp/$pc_rpm", 1000 );
+
+
+    #Create repo
+    $instance->run_ssh_command(cmd => "sudo zypper in -y inst-source-utils", proceed_on_failure => 0);
+    $instance->ssh_script_run( cmd => "mkdir -p $tmp_repo $sle15_mig_repo; cd $tmp_repo");
+    $instance->ssh_script_run( cmd => "create_update_source.sh .");
+    $instance->ssh_script_run( cmd => "cd $tmp_repo/updates; mkdir -p $tmp_repo/updates/suse/noarch");
+    $instance->ssh_script_run( cmd => "cp -av /tmp/$dms_rpm_name $tmp_repo/updates/suse/noarch/.; cp -av /tmp/$pc_rpm $tmp_repo/updates/suse/noarch/.");
+    $instance->ssh_script_run( cmd => "cd $tmp_repo/updates/suse");
+    $instance->ssh_script_run( cmd => "create_package_descr -x setup/descr/EXTRA_PROV -C");
+    $instance->ssh_script_run( cmd => "cd $tmp_repo/updates; sudo tar -czvf ../sles15-mig-repo.tgz *; cp $tmp_repo/sles15-mig-repo.tgz $sle15_mig_repo/.");
+    $instance->ssh_script_run( cmd => "cd $sle15_mig_repo; tar -xzvf $sle15_mig_repo/sles15-mig-repo.tgz -C $sle15_mig_repo");
+    $instance->ssh_script_run( cmd => "sudo zypper addrepo $sle15_mig_repo SLE15-Migration");
+
+    #Refresh and Update
+    $instance->ssh_assert_script_run( 'sudo zypper ref; sudo zypper -n up',
+        timeout => 300 );
 
     # Upload distro_migration.log
     $instance->upload_log("/system-root/var/log/distro_migration.log", failok => 1);
 
-    $instance->ssh_script_run(cmd => "sudo zypper -n in SLES15-Migration suse-migration-sle15-activation", timeout => 300);
+    record_info('installs SLE15-Migration and suse-migration-sle15-activation');
```

Comment:
you stating that you installing the package but actually you don't , you importing repo key it may be confusing for review

Reply:
record_info("Import $dms_repo_key");
    $instance->run_ssh_command(
        cmd                => "sudo rpm --import /tmp/$dms_repo_key",
        proceed_on_failure => 0
    );
    record_info("installs SLE15-Migration and $act_rpm");
    $instance->run_ssh_command(
        cmd                => "sudo zypper in -y SLE15-Migration /tmp/$act_rpm",
        proceed_on_failure => 0
    );

Fixed it